scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,37 @@
+cimport cython
+from numpy cimport int64_t, intp_t
+import numpy as np
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _kendall_condis(intp_t[:] x, intp_t[:] y):
+    cdef:
+        intp_t sup = 1 + np.max(y)
+        intp_t[::1] arr = np.zeros(sup, dtype=np.intp)
+        intp_t i = 0, k = 0, size = x.size, idx
+        int64_t con = 0, dis = 0
+
+    while i < size:
+        while k < size and x[i] == x[k]:
+            idx = y[k] - 1
+            while idx != 0:
+                con += arr[idx]
+                idx -= idx & -idx
```

Comment:
Note that the following code

    x = x & (x-1)

will delete the lowest bit with one operation less than

    x -= x & -x

It is used in all standard implementations of Fenwick trees I've seen.